include/aarch64_multibinary.h

```diff
@@ -168,32 +182,65 @@
  */
 .macro mbin_interface_base name:req, base:req
 	.extern \base
-	.section        .data
+	.data
 	.balign 8
-	.global \name\()_dispatcher_info
+	.global cdecl(\name\()_dispatcher_info)
+#ifndef __APPLE__
 	.type   \name\()_dispatcher_info,%object
-
-	\name\()_dispatcher_info:
+#endif
+	cdecl(\name\()_dispatcher_info):
 		.quad   \base         //func_entry
+#ifndef __APPLE__
 	.size   \name\()_dispatcher_info,. - \name\()_dispatcher_info
-
+#endif
 	.balign 8
 	.text
-	.global \name
+	.global cdecl(\name)
+#ifndef __APPLE__
 	.type \name,%function
+#endif
 	.align  2
-	\name\():
-		adrp    x9, :got:\name\()_dispatcher_info
-		ldr     x9, [x9, #:got_lo12:\name\()_dispatcher_info]
+	cdecl(\name\()):
+#ifndef __APPLE__
+		adrp    x9, :got:cdecl(_\name\()_dispatcher_info)
+		ldr     x9, [x9, #:got_lo12:cdecl(_\name\()_dispatcher_info)]
+#else
+		adrp    x9, cdecl(_\name\()_dispatcher_info)@GOTPAGE
+		ldr     x9, [x9, #cdecl(_\name\()_dispatcher_info)@GOTPAGEOFF]
+#endif
 		ldr     x10,[x9]
 		br      x10
+#ifndef __APPLE__
 	.size \name,. - \name
-
+#endif
 .endm
 
 #else /* __ASSEMBLY__ */
+#include <stdint.h>
+#if defined(__linux__)
 #include <sys/auxv.h>
-
+#elif defined(__APPLE__)
+int _get_cpu_capabilities(void);
+#define kHasICDSBShift                  2
```

Comment:
This part looks same with the code in  https://opensource.apple.com/source/xnu/xnu-6153.81.5/osfmk/arm/cpu_capabilities.h.auto.html .
It is release under http://www.opensource.apple.com/apsl/ . 

I am not sure if it is conflict with BSD license . 
@gbtucker  please review .